File: sdks/go/models/background_task_out.go
package models

import "time"

type BackgroundTaskOut struct {
	Id         string                 `json:"id"`
	CustomerId string                 `json:"customerId"`
	TaskType   string                 `json:"taskType"`
	Status     string                 `json:"status"`
	Data       map[string]interface{} `json:"data,omitempty"`
	Result     map[string]interface{} `json:"result,omitempty"`
	Error      *string                `json:"error,omitempty"`
	Progress   int16                  `json:"progress"`
	CreatedAt  time.Time              `json:"createdAt"`
	StartedAt  *time.Time             `json:"startedAt,omitempty"`
	FinishedAt *time.Time             `json:"finishedAt,omitempty"`
}

File: sdks/java/lib/src/main/java/com/hooksniff/models/BackgroundTaskOut.java
package com.hooksniff.models;

import com.fasterxml.jackson.annotation.JsonProperty;

public class BackgroundTaskOut {
    public String id;
    @JsonProperty("customer_id") public String customerId;
    @JsonProperty("task_type") public String taskType;
    public String status;
    public Object data;
    public Object result;
    public String error;
    public int progress;
    @JsonProperty("created_at") public String createdAt;
    @JsonProperty("started_at") public String startedAt;
    @JsonProperty("finished_at") public String finishedAt;
}
